src/axom/sidre/tests/sidre_read_write_userdefined_data.cpp

```diff
@@ -0,0 +1,706 @@
+}
+
+#endif  // AXOM_USE_MFEM
+
+//------------------------------------------------------------------------------
+
+TEST(sidre, OneD_double_readandwrite) { test_user_defined_data<double, 1>(); }
+
+TEST(sidre, OneD_StateOne_readandwrite)
+{
+  test_user_defined_data<StateOne, 1>();
+}
+
+TEST(sidre, OneD_StateTwo_readandwrite)
+{
+  test_user_defined_data<StateTwo, 1>();
+}
+
+TEST(sidre, OneD_StateThree_readandwrite)
+{
+  test_user_defined_data<StateThree, 1>();
+}
+
+TEST(sidre, OneD_StateTensorSmall_readandwrite)
+{
+  test_user_defined_data<StateTensorSmall, 1>();
+}
+
+TEST(sidre, OneD_StateTensorLarge_readandwrite)
+{
+  test_user_defined_data<StateTensorLarge, 1>();
+}
+
+//-------------------------
+
+TEST(sidre, TwoD_double_readandwrite) { test_user_defined_data<double, 2>(); }
+
+TEST(sidre, TwoD_StateOne_readandwrite)
+{
+  test_user_defined_data<StateOne, 2>();
+}
+
+TEST(sidre, TwoD_StateTwo_readandwrite)
+{
+  test_user_defined_data<StateTwo, 2>();
+}
+
+TEST(sidre, TwoD_StateThree_readandwrite)
+{
+  test_user_defined_data<StateThree, 2>();
+}
+
+TEST(sidre, TwoD_StateTensorSmall_readandwrite)
+{
+  test_user_defined_data<StateTensorSmall, 2>();
+}
+
+TEST(sidre, TwoD_StateTensorLarge_readandwrite)
+{
+  test_user_defined_data<StateTensorLarge, 2>();
+}
+
+//------------------------------------------------------------------------------
+
+TEST(sidre, OneD_double_external_readandwrite)
+{
+  test_external_user_defined_data<double, 1>();
+}
+
+TEST(sidre, OneD_StateOne_external_readandwrite)
+{
+  test_external_user_defined_data<StateOne, 1>();
+}
+
+TEST(sidre, OneD_StateTwo_external_readandwrite)
+{
+  test_external_user_defined_data<StateTwo, 1>();
+}
+
+TEST(sidre, OneD_StateThree_external_readandwrite)
+{
+  test_external_user_defined_data<StateThree, 1>();
+}
+
+TEST(sidre, OneD_StateTensorSmall_external_readandwrite)
+{
+  test_external_user_defined_data<StateTensorSmall, 1>();
+}
+
+TEST(sidre, OneD_StateTensorLarge_external_readandwrite)
+{
+  test_external_user_defined_data<StateTensorLarge, 1>();
+}
+
+//-------------------------
+
+TEST(sidre, TwoD_double_external_readandwrite)
+{
+  test_external_user_defined_data<double, 2>();
+}
+
+TEST(sidre, TwoD_StateOne_external_readandwrite)
+{
+  test_external_user_defined_data<StateOne, 2>();
+}
+
+TEST(sidre, TwoD_StateTwo_external_readandwrite)
+{
+  test_external_user_defined_data<StateTwo, 2>();
+}
+
+TEST(sidre, TwoD_StateThree_external_readandwrite)
+{
+  test_external_user_defined_data<StateThree, 2>();
+}
+
+TEST(sidre, TwoD_StateTensorSmall_external_readandwrite)
+{
+  test_external_user_defined_data<StateTensorSmall, 2>();
+}
+
+TEST(sidre, TwoD_StateTensorLarge_external_readandwrite)
+{
+  test_external_user_defined_data<StateTensorLarge, 2>();
+}
+
+//-------------------------
+
+#ifdef AXOM_USE_MFEM
+
+TEST(sidre, OneD_double_MFEMSidreDataCollection_readandwrite)
+{
+  test_MFEMSidreDataCollection_user_defined_data<double, 1>();
+}
+/*
```

Comment:
Please add a comment about why these are commented out, and add or update an issue to fix these when SPIO can load into the specified group in the `MFEMSidreDataCollection`

Reply:
I completely forgot I wrote these then wrote a much simpler single test in the MFEMSidreDataCollection test file..... good catch!  They even passed just uncommenting them. Past Me was great!